### PR TITLE
fix: ensure config changed called on trusted-changed

### PR DESCRIPTION
**Pull request by marcoppenheimer**
##### `feat: ensure config-changed updates properties during trusted-relation-changed events`
- https://warthogs.atlassian.net/browse/DPE-1985
##### `docs: add additional characters and correct quoting to ssl_principal_mapping_rules example`
- https://warthogs.atlassian.net/browse/DPE-1987
##### `feat: enable ssl jmx debug server logs by default`
- https://warthogs.atlassian.net/browse/DPE-1986

### CODE REVIEW

**Review by zmraul — CHANGES_REQUESTED** [review]
It's great that you tracked this issue! I have a thought about the cause.

File: src/tls.py [hunk 1]
```diff
@@ -201,8 +201,17 @@ def _trusted_relation_changed(self, event: RelationChangedEvent) -> None:
         safe_write_to_file(content=content, path=f"{self.charm.snap.CONF_PATH}/{filename}")
         self.import_cert(alias=f"{alias}", filename=filename)
 
+        # ensuring new config gets applied
+        self.charm._on_config_changed(event)
```
[the remark this answers is not in the document]
Comment (deusebio) [reply src/tls.py:205]: Uhm. Good point. Also, we should not rely on changing mtls="enable" only to trigger update of truststore. As we could have mulitple relations of `trusted-certificate` and `trusted-ca`, and only the first one would trigger a change in the peer field `mtls="enabled"`

Let's sync today after the daily call, to get to the bottom of this!

**Review by zmraul — APPROVED** [review]

**Review by deusebio — APPROVED** [review]
LGTM!

File: src/config.py [hunk 2]
```diff
@@ -212,6 +212,7 @@ def jmx_opts(self) -> str:
         opts = [
             "-Dcom.sun.management.jmxremote",
             f"-javaagent:{self.jmx_prometheus_javaagent_filepath}={JMX_EXPORTER_PORT}:{self.jmx_prometheus_config_filepath}",
+            "-Djavax.net.debug=ssl:handshake:verbose:session:keymanager:trustmanager",
```
[the remark this answers is not in the document]
Comment (marcoppenheimer) [reply src/config.py:215]: Will do in other PR.

**Pull request merged**